test/inductor/test_distributed_patterns.py

```diff
@@ -13,76 +12,100 @@
 
 # Fake distributed
 WORLD_SIZE = 2
-RESIZE = True
 
 
-def init_fake_distributed():
+def init_fake_distributed(device="cpu"):
     @torch.no_grad
     def all_gather(t):
-        return torch.cat([t] * WORLD_SIZE, 0)
+        # clone since all_gather input and output should not be aliases.
+        return torch.cat([t] * WORLD_SIZE, 0).clone()
 
     @torch.no_grad
     def reduce_scatter(t):
-        return t.narrow(0, 0, t.size(0) // WORLD_SIZE)
+        # clone since reduce_scatter input and output should not be aliases.
+        return t.narrow(0, 0, t.size(0) // WORLD_SIZE).clone()
 
     def fw_pre_hook(mod, inp):
-        with torch.no_grad():
-            mod.og_weight = mod.weight
-            mod.weight = nn.Parameter(all_gather(mod.weight))
+        if not compiled_autograd.compiled_autograd_enabled:
+            # torch.ops.fsdp.set_ doesn't work well in eager mode, so use the slow copy_ path instead.
```

Comment:
How does `fsdp.set_` break in eager mode? (now that your PR adds a cpu and cuda kernel for `set_`, in `_fsdp_param.py`)

Totally fine if you want to stick with `copy_` - although I think you will end up going down the "two inputs were aliased and one was mutated" path in AOTAutograd unnecessarily.

Reply:
yes it's still a little hard to figure out why `set_` fails in eager mode - it throws `setStorage: sizes [10, 20], strides [20, 1], storage offset 0, and itemsize 4 requiring a storage size of 800 are out of bounds for storage of size 0` but doesn't tell which op it is failing on.

maybe for now we can stick with `copy_` for eager mode, and this won't cause problem with AOTAutograd because in compile mode we use `set_` instead which works.